File: MLR-spring/src/main/java/com/mlr/entity/Transaction.java
package com.mlr.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import java.time.LocalDateTime;

@Entity
@Table(name = "transactions")
@Getter @Setter @NoArgsConstructor
public class Transaction {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "user_id")
    private String userId;

    private Double amount;

    // 0 = Credit, 1 = Debit
    private int type;

    // 0=Salary, 1=Cash, 2=Refund, 3=Payment
    private int source;

    private String description;

    private String merchant;

    private String upiId; 

    @Column(name = "category_id")
    private Integer categoryId;

    private LocalDateTime createdAt = LocalDateTime.now();

 
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", insertable = false, updatable = false)
    private User user;

    @ManyToOne(fetch = FetchType.EAGER) // Eager load category name for history
    @JoinColumn(name = "category_id", insertable = false, updatable = false)
    private Category category;
}

File: MLR/DAO/Entities/Transaction.cs
using System.ComponentModel.DataAnnotations;

namespace MLR.DAO;

public class Transaction
{
    [Key]
    public int Id { get; set; }

    public string UserId { get; set; } = string.Empty;

    public decimal Amount { get; set; }

    // 0 = Credit, 1 = Debit
    public int Type { get; set; } = 0;

    // 0=Salary, 1=Cash, 2=Refund, 3=Payment
    public int Source { get; set; } = 0;

    public string Description { get; set; } = string.Empty;

    public string? Merchant { get; set; }

    // Added UPI ID field
    public string? UpiId { get; set; }

    public int CategoryId { get; set; }

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public virtual ApplicationUser User { get; set; } = null!;
    public virtual Category Category { get; set; } = null!;
}